File: debug-parse.py
import re

with open(r'C:/xampp/htdocs/florieren/nestjs-backend/greatkin_gk.sql', 'r', encoding='utf-8') as f:
    sql = f.read()

def parse(table):
    pat = r'INSERT INTO `' + re.escape(table) + r'`\s*\(([^)]+)\)\s*VALUES\s*([\s\S]*?);(?=\s*--|\s*INSERT|\s*CREATE|\s*$)'
    m = re.search(pat, sql, re.IGNORECASE)
    print(f'Table {table}: match={m is not None}')
    if not m:
        return [], []
    cols = [c.strip().replace('`', '') for c in m.group(1).split(',')]
    print(f'  Cols: {cols}')
    vb = m.group(2).strip()
    print(f'  Values block length: {len(vb)}')
    rows = []
    cur = ''
    ins = False
    sc = "'"
    for ch in vb:
        if ins:
            if ch == sc and cur[-1] != '\\':
                ins = False
            cur += ch
        else:
            if ch == "'" or ch == '"':
                ins = True
                sc = ch
                cur = ch
            elif ch == '(':
                cur = ''
            elif ch == ')' and cur.strip():
                rows.append(cur.strip())
                cur = ''
            else:
                cur += ch
    print(f'  Raw rows: {len(rows)}')
    out = []
    for rs in rows[:3]:
        print(f'  Row sample: {rs[:100]}')
    return cols, rows[:3]

parse('staff')
parse('users')

File: debug-parse.js
const fs = require('fs');
const sql = fs.readFileSync('C:/xampp/htdocs/florieren/nestjs-backend/greatkin_gk.sql', 'utf8');

function parseSqlRows(sql, tableName) {
  // Find INSERT INTO statement
  const regex = new RegExp('INSERT INTO `' + tableName + '`\\s*\\(([^)]+)\\)\\s*VALUES', 'i');
  const match = sql.match(regex);
  if (!match) return { cols: [], rows: [] };
  
  const cols = match[1].split(',').map(c => c.trim().replace(/`/g, ''));
  
  // Find VALUES block - it might span multiple INSERT statements
  // Actually, let's find the first VALUES block after the header
  const valuesIdx = sql.indexOf('VALUES', match.index + match[0].length);
  if (valuesIdx === -1) return { cols, rows: [] };
  
  const valuesBlock = sql.substring(valuesIdx + 6);
  
  const rows = [];
  let cur = '', inStr = false, sc = '';
  for (let i = 0; i < valuesBlock.length; i++) {
    const ch = valuesBlock[i];
    if (inStr) {
      if (ch === sc && valuesBlock[i-1] !== '\\') inStr = false;
      cur += ch;
    } else {
      if (ch === "'" || ch === '"') { inStr = true; sc = ch; cur += ch; }
      else if (ch === '(') {
        if (cur.trim()) rows.push(cur.trim());
        cur = '';
      }
      else if (ch === ')' && cur.trim()) {
        cur += ch;
        rows.push(cur.trim());
        cur = '';
      }
      else if (ch === ';' && cur.trim() === '') {
        break;
      }
      else cur += ch;
    }
  }
  
  const parsedRows = [];
  for (const rowStr of rows) {
    const vals = [];
    let rCur = '', rInStr = false, rSc = '';
    for (let i = 0; i < rowStr.length; i++) {
      const ch = rowStr[i];
      if (rInStr) {
        if (ch === rSc && rowStr[i-1] !== '\\') rInStr = false;
        rCur += ch;
      } else {
        if (ch === "'" || ch === '"') { rInStr = true; rSc = ch; rCur += ch; }
        else if (ch === ',') { vals.push(rCur.trim()); rCur = ''; }
        else rCur += ch;
      }
    }
    if (rCur.trim()) vals.push(rCur.trim());
    if (vals.length > 0) parsedRows.push(vals);
  }
  
  return { cols, rows: parsedRows };
}

function getSrc(sql, tableName) {
  return parseSqlRows(sql, tableName);
}

const { cols: staffCols, rows: staffRows } = getSrc(sql, 'staff');
console.log('Staff cols:', staffCols.join(', '));
console.log('Staff rows:', staffRows.length);
console.log('First staff row:', staffRows[0].join(', '));
console.log('First staff unique_id:', staffRows[0][staffCols.indexOf('unique_id')]);

const { cols: userCols, rows: userRows } = getSrc(sql, 'users');
console.log('\nUsers cols:', userCols.join(', '));
console.log('Users rows:', userRows.length);
console.log('First user id:', userRows[0][userCols.indexOf('student_id')]);

const { cols: adminCols, rows: adminRows } = getSrc(sql, 'admin');
console.log('\nAdmin cols:', adminCols.join(', '));
console.log('Admin rows:', adminRows.length);
console.log('First admin unique_id:', adminRows[0][adminCols.indexOf('unique_id')]);
